examples/gno.land/r/demo/userbook/userbook.gno

```diff
@@ -0,0 +1,153 @@
+// This realm demonstrates a small userbook system working with gnoweb
+package userbook
+
+import (
+	"std"
+	"strconv"
+
+	"gno.land/p/demo/avl"
+	"gno.land/p/demo/mux"
+	"gno.land/p/demo/ufmt"
+)
+
+type Signup struct {
+	account string
+	height  int64
+}
+
+// signups - keep a slice of signed up addresses efficient pagination
+var signups []Signup
+
+// tracker - keep track of who signed up
+var (
+	tracker *avl.Tree
+	router  *mux.Router
+)
+
+const (
+	defaultPageSize = 20
+	pathArgument    = "number"
+	subPath         = "page/{" + pathArgument + "}"
+)
+
+func init() {
+	// Set up tracker tree
+	tracker = avl.NewTree()
+
+	// Set up route handling
+	router = mux.NewRouter()
+	router.HandleFunc("", renderHelper)
+	router.HandleFunc(subPath, renderHelper)
+
+	// Sign up the deployer
+	SignUp()
+}
+
+func SignUp() string {
+	// Get transaction caller
+	caller := std.PrevRealm().Addr().String()
+	height := std.GetHeight()
+
+	if _, exists := tracker.Get(caller); exists {
+		panic(caller + " is already signed up!")
+	}
+
+	tracker.Set(caller, struct{}{})
+	signup := Signup{
+		caller,
+		height,
+	}
+
+	signups = append(signups, signup)
+	return ufmt.Sprintf("%s added to userbook up at block #%d!", signup.account, signup.height)
+}
+
+func GetSignupsInRange(page, pageSize int) ([]Signup, int) {
+	if page < 1 {
+		panic("page number cannot be less than 1")
+	}
```

Comment:
Okay for now, but please switch to p/pagination (or eq.) as soon as it's available.